src/Compilers/Core/Portable/Diagnostic/LocalizableString.cs

```diff
@@ -45,29 +43,61 @@ public override bool Equals(object other)
             return Equals(other as LocalizableString);
         }
 
-        private sealed class FixedLocalizableString : LocalizableString
+        internal LocalizableString MakeExceptionSafe()
```

Comment:
can this get onException as an argument rather than using SetException .... to set field?